exdb/views.py

```diff
@@ -368,6 +368,97 @@ def get_context_data(self, *args, **kwargs):
         return context
 
 
+class CompletionBoardView(TemplateView):
+    access_level = 'basic'
+    template_name = 'exdb/completion_board.html'
+
+    def get_context_data(self, *args, **kwargs):
+        context = super(CompletionBoardView, self).get_context_data(*args, **kwargs)
+
+        if not self.request.user.is_hallstaff():
+            raise Http404('User does not have access to the global completion board')
+
+        affiliation_pk = self.kwargs.get('pk') or self.request.user.affiliation_id
+        if affiliation_pk is not None:
+            affiliation = get_object_or_404(Affiliation, pk=affiliation_pk)
+        else:
+            # Just get the first one for now
+            affiliation = Affiliation.objects.first()
```

Comment:
The reason for this was so that we could optionally exclude the PK from the completion board links and still have a reasonable default if the user doesn't have an Affiliation set on their user object.